sentry-android-core/src/main/java/io/sentry/android/core/util/CheckConnectivity.java

```diff
@@ -0,0 +1,54 @@
+package io.sentry.android.core.util;
+
+import android.Manifest;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import io.sentry.core.ILogger;
+import io.sentry.core.SentryLevel;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class CheckConnectivity {
+
+  private CheckConnectivity() {}
```

Comment:
What about we name the class like:
```suggestion
  private ConectivityChecker() {}
```

Or `ConectivitySomethingElse` since `CheckConectivity` sounds like an action or method

Reply:
yeah, it makes sense, sometimes I feel writing extension methods hehe